src/routing_table/xorable.rs

```diff
@@ -128,6 +162,22 @@ macro_rules! impl_xorable {
                 let pow_i = 1 << (mem::size_of::<Self>() * 8 - 1 - i); // 1 on bit i.
                 self ^ pow_i
             }
+
+            fn binary(&self) -> String {
+                let bits = mem::size_of::<Self>() * 8;
+                let binary = format!("{:1b}", self);
```

Comment:
It says in the `fmt` module documentation that the `0` after the `:` will make it pad with zeros instead of spaces.

And your second point is what I meant by `match`ing against the size of `Self`: each branch would use the appropriate width, e.g. `{:064b}` for `u64`.